File: commands/clean.js
const chalk = require('chalk');
const fs = require('fs');
const path = require('path');
const { parseRepo } = require('../lib/git');

/**
 * Clean cached repositories
 */
async function clean(repo, options) {
  try {
    const cacheDir = options.cacheDir;

    if (repo) {
      // Clean specific repository
      const { owner, name } = parseRepo(repo);
      const repoPath = path.join(cacheDir, owner, name);

      if (fs.existsSync(repoPath)) {
        fs.rmSync(repoPath, { recursive: true, force: true });
        console.log(chalk.green(`✅ Cleaned cache for ${repo}`));
      } else {
        console.log(chalk.yellow(`⚠️  No cache found for ${repo}`));
      }
    } else {
      // Clean all cache
      if (options.all) {
        if (fs.existsSync(cacheDir)) {
          const size = getDirectorySize(cacheDir);
          fs.rmSync(cacheDir, { recursive: true, force: true });
          console.log(chalk.green(`✅ Cleaned all cache (${formatBytes(size)})`));
        } else {
          console.log(chalk.yellow('⚠️  No cache found'));
        }
      } else {
        console.log(chalk.yellow('⚠️  Use --all flag to clean all cached repositories'));
        console.log(chalk.gray('   Or specify a repository: ghr clean owner/repo'));
      }
    }

    // Clean output directory if requested
    if (options.output) {
      const outputDir = options.outputDir;
      if (fs.existsSync(outputDir)) {
        fs.rmSync(outputDir, { recursive: true, force: true });
        console.log(chalk.green('✅ Cleaned output directory'));
      }
    }

  } catch (error) {
    console.error(chalk.red(`\n❌ Error: ${error.message}`));
    process.exit(1);
  }
}

/**
 * Get directory size recursively
 */
function getDirectorySize(dirPath) {
  let size = 0;

  function calculateSize(filePath) {
    const stats = fs.statSync(filePath);
    if (stats.isDirectory()) {
      const files = fs.readdirSync(filePath);
      files.forEach(file => {
        calculateSize(path.join(filePath, file));
      });
    } else {
      size += stats.size;
    }
  }

  if (fs.existsSync(dirPath)) {
    calculateSize(dirPath);
  }

  return size;
}

/**
 * Format bytes to human readable
 */
function formatBytes(bytes) {
  if (bytes === 0) {return '0 B';}
  const k = 1024;
  const sizes = ['B', 'KB', 'MB', 'GB'];
  const i = Math.floor(Math.log(bytes) / Math.log(k));
  return Math.round((bytes / Math.pow(k, i)) * 100) / 100 + ' ' + sizes[i];
}

module.exports = { clean };

File: cmd/clean.go
package cmd

import (
	"fmt"
	"os"
	"path/filepath"

	"github.com/spf13/cobra"
	"github.com/syxc/ghr/internal/git"
	"github.com/syxc/ghr/internal/utils"
)

var (
	cleanAll    bool
	cleanOutput bool
)

// cleanCmd represents the clean command
var cleanCmd = &cobra.Command{
	Use:   "clean [repo]",
	Short: "Clean cached repositories",
	Long:  `Clean cached repositories from the local cache directory.`,
	Args:  cobra.MaximumNArgs(1),
	RunE: func(cmd *cobra.Command, args []string) error {
		repo := ""
		if len(args) > 0 {
			repo = args[0]
		}
		return runClean(repo)
	},
}

func init() {
	rootCmd.AddCommand(cleanCmd)
	cleanCmd.Flags().BoolVarP(&cleanAll, "all", "a", false, "Clean all cached repositories")
	cleanCmd.Flags().BoolVarP(&cleanOutput, "output", "o", false, "Clean output directory as well")
}

func runClean(repo string) error {
	cfg := getConfig()

	if repo != "" {
		// Clean specific repository
		owner, name, err := git.ParseRepo(repo)
		if err != nil {
			utils.PrintError("%v", err)
			return err
		}

		repoPath := filepath.Join(cfg.CacheDir, owner, name)
		if _, err := os.Stat(repoPath); err == nil {
			if err := os.RemoveAll(repoPath); err != nil {
				utils.PrintError("%v", err)
				return err
			}
			fmt.Println(utils.Green(fmt.Sprintf("✅ Cleaned cache for %s", repo)))
		} else {
			fmt.Println(utils.Yellow(fmt.Sprintf("⚠️  No cache found for %s", repo)))
		}
	} else {
		// Clean all cache
		if cleanAll {
			if _, err := os.Stat(cfg.CacheDir); err == nil {
				size, _ := utils.GetDirectorySize(cfg.CacheDir)
				if err := os.RemoveAll(cfg.CacheDir); err != nil {
					utils.PrintError("%v", err)
					return err
				}
				fmt.Println(utils.Green(fmt.Sprintf("✅ Cleaned all cache (%s)", utils.FormatBytes(size))))
			} else {
				fmt.Println(utils.Yellow("⚠️  No cache found"))
			}
		} else {
			fmt.Println(utils.Yellow("⚠️  Use --all flag to clean all cached repositories"))
			fmt.Println(utils.Gray("   Or specify a repository: ghr clean owner/repo"))
		}
	}

	// Clean output directory if requested
	if cleanOutput {
		if _, err := os.Stat(cfg.OutputDir); err == nil {
			if err := os.RemoveAll(cfg.OutputDir); err != nil {
				utils.PrintError("%v", err)
				return err
			}
			fmt.Println(utils.Green("✅ Cleaned output directory"))
		}
	}

	return nil
}
